service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -77,21 +82,122 @@
+                }
+            }));
+    }
+
+    /**
+     * Adds a response by an assignee to a direction where the direction ID of the response matches the direction ID
+     * of the direction.
+     *
+     * @param responses  a list of single responses to directions.
+     * @param directions a list of directions with combined responses.
+     */
+    public void addResponsesToDirections(List<DirectionResponse> responses, List<Element<Direction>> directions) {
+        directions.forEach(direction -> responses.stream()
+            .filter(response -> response.getDirectionId().equals(direction.getId()))
+            .forEach(response -> {
+                direction.getValue().getResponses().removeIf(x ->
+                    responseExists(response, direction) && x.getValue().getAssignee().equals(response.getAssignee()));
+
+                direction.getValue().getResponses()
+                    .add(Element.<DirectionResponse>builder()
+                        .id(UUID.randomUUID())
+                        .value(response)
+                        .build());
+            }));
+    }
+
+    private boolean responseExists(DirectionResponse response, Element<Direction> direction) {
+        return isNotEmpty(direction.getValue().getResponses()) && response.getDirectionId().equals(direction.getId())
+            && direction.getValue().getResponses().stream()
+            .anyMatch(directionResponse -> directionResponse.getValue().getAssignee().equals(response.getAssignee()));
+    }
+
+    /**
+     * Extracts a specific response to a direction by a party.
+     *
+     * @param assignee   the role that responded to the direction.
+     * @param directions a list of directions.
+     * @return a list of directions with the correct response associated.
+     */
+    public List<Element<Direction>> extractPartyResponse(String assignee, List<Element<Direction>> directions) {
+        return directions.stream()
+            .map(element -> Element.<Direction>builder()
+                .id(element.getId())
+                .value(element.getValue().toBuilder()
+                    .response(element.getValue().getResponses().stream()
+                        .filter(response -> response.getValue().getDirectionId().equals(element.getId()))
+                        .filter(response -> response.getValue().getAssignee().getValue().equals(assignee))
+                        .map(Element::getValue)
+                        .findFirst()
+                        .orElse(null))
+                    .build())
+                .build())
+            .collect(toList());
+    }
+
+    /**
+     * Collects directions for all parties and places them into single map, where the key is the role direction,
+     * and the value is the list of directions associated.
+     *
+     * @param caseData data from case.
+     * @return Map of roles and directions.
+     */
+    public Map<String, List<Element<Direction>>> collectDirectionsToMap(CaseData caseData) {
```

Comment:
Does the map key need to be a `String` can it not be a `DirectionAssignee`? Seems the only place that it is used is in `addHiddenVariablesToResponseForAssignee` where you are then converting the `String` back into a `DirectionAssignee`.

Reply:
this is a good spot, it would actually totally remove the need for my string -> DirectionAssignee method I think 🤔

Reply:
I think so as well